docs/rfcs/2023-11-01-wagon-file-format.md

```diff
@@ -0,0 +1,77 @@
+---
+Feature Name: Wagon File Format
+Tracking Issue: TBD
+Date: 2023-11-01
+Author: "Zhong Zhenchi <zhongzc_arch@outlook.com>"
+---
+
+## Abstract 
+
+This document presents a new design, the Wagon file format, created to replace [Iceberg's Puffin format](https://iceberg.apache.org/puffin-spec/) that is implemented in Java and thus not directly usable in our project. The Wagon file format caters for various types of data, with a primary focus on index data.
```

Comment:
This format is quite close to Puffin (almost a subset of it). Can we port the Puffin format to Rust? We don't need to support all features (e.g. compression). The benefit is that Java users are able to use Puffin's Java API to read our index files.

Reply:
You're right, it does look a lot like a subset of Puffin. Also, implementing Puffin in Rust certainly presents no challenges. The idea of allowing users to read our index file via the Java API is a perspective and advantage that hadn't crossed my mind - it's an excellent suggestion.